### PR TITLE
allow networks to be created from Boolean logic text file

### PR DESCRIPTION
**Pull request by tjparmer**
Included changes to allow the network to be read in from a Boolean logic text file (as used by Reka Albert's lab), exemplar leukemia and breast cancer text files, and updates to the breast cancer tutorial to reformat the effective graph to mimic Figure 3 from the respective paper

### CODE REVIEW

File: cana/utils.py [hunk 1]
```diff
@@ -293,3 +293,26 @@ def isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
...
+def output_transitions(eval_line,input_list):
+	'''
+	returns an output list from combinatorically trying all input values
+	Args:
+	    eval_line (string) : logic or arithmetic line to evaluate
+	    input_list (list) : list of input variables
+
+	Returns:
+		list of all possible output transitions
+	'''
```
Comment (rionbr) [inline cana/utils.py]: Thomas, can we have an example in the docstring of what this functions does?

File: cana/datasets/leukemia.txt [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#BOOLEAN RULES
```
Comment (rionbr) [inline cana/datasets/leukemia.txt:5]: We need to add a line of comment before the file start with the reference from the paper. I believe some of the other files also have that. Take a look at them.

File: cana/boolean_network.py [hunk 3]
```diff
@@ -151,6 +152,58 @@ def from_string(cls, input_string, keep_constants=True, **kwargs):
 
 		return cls.from_dict(logic, keep_constants=keep_constants, **kwargs)
 
+	@classmethod
+	def from_text(cls, input_file, keep_constants=True, **kwargs):
```
Comment (rionbr) [inline cana/boolean_network.py]: So now we have two function "from_text" and "from_string" .. It is not intuitive that they do different things. I believe "from_text" deals with Albert format, "from_string" deals with cnet format.

Perhaps we should have one function called "from_string" with a parameter that then calls other functions depending on the format. Let's talks about this more.

**Review by rionbr — CHANGES_REQUESTED** [review]

**Comment by tjparmer** [comment]
I've addressed the three points you mentioned above.  I have also created a keyword argument for from_file so that it can accept either cnet format or Albert's boolean rules format.  The default is cnet which calls from_string, so that the older code doesn't break.  I tested with the old networks (Thaliana, drosophila) and the new cancer networks to make sure they still work.

**Review by rionbr — APPROVED** [review]

**Comment by rionbr** [comment]
Self note: I wanted to change some things on @tjparmer pull request before merging. I wasn't able to do that, so now I've merged and will update some variables.

**Pull request merged**